pkg/azureutils/informer_common.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azureutils
+
+import (
+	"context"
+
+	crdInformers "k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+	azdiskinformers "sigs.k8s.io/azuredisk-csi-driver/pkg/apis/client/informers/externalversions"
+)
+
+type InformerFactory interface {
+	Start(stopCh <-chan struct{})
+	Informer() cache.SharedIndexInformer
+}
+
+type azNodeInformer struct {
+	factory  azdiskinformers.SharedInformerFactory
+	informer cache.SharedIndexInformer
+}
+
+func NewAzNodeInformer(factory azdiskinformers.SharedInformerFactory) InformerFactory {
+	return &azNodeInformer{
+		factory:  factory,
+		informer: factory.Disk().V1beta2().AzDriverNodes().Informer(),
+	}
+}
+
+func (a azNodeInformer) Start(stopCh <-chan struct{}) {
```

Comment:
I'm not sure should we use `*azNodeInformer` here, like: `func (a *azNodeInformer) Start(stopCh <-chan struct{}) {`

Reply:
Good catch! Yes, the receivers should be pointers instead of by value.